tests/mdev_configuration_allocation_test.go

```diff
@@ -74,6 +53,33 @@ var _ = Describe("[Serial][sig-compute]MediatedDevices", func() {
 		}
 	}
 
+	checkAllMDEVCreated := func(mdevTypeName string, expectedInstancesCount int) func() (*k8sv1.Pod, error) {
+		return func() (*k8sv1.Pod, error) {
+			By(fmt.Sprintf("Checking the number of created mdev types, should be %d of %s type ", expectedInstancesCount, mdevTypeName))
+			check := fmt.Sprintf(`set -x
+		files_num=$(ls -A /sys/bus/mdev/devices/| wc -l)
+		if [[ $files_num != %d ]] ; then
+		  echo "failed, not enough mdevs of type %[2]s has been created"
+		  exit 1
+		fi
+		for x in $(ls -A /sys/bus/mdev/devices/); do
+		  type_name=$(basename "$(readlink -f /sys/bus/mdev/devices/$x/mdev_type)")
+		  if [[ "$type_name" != "%[2]s" ]]; then
+		     echo "failed, not all mdevs of type %[2]s"
+		     exit 1
+		  fi
+		  exit 0
+		done`, expectedInstancesCount, mdevTypeName)
+			testPod := tests.RenderPod("test-all-mdev-created", []string{"/bin/bash", "-c"}, []string{check})
+			testPod, err = virtClient.CoreV1().Pods(util.NamespaceTestDefault).Create(context.Background(), testPod, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
can we avoid this expect within the helper ? i.e. return the error ? 

If not, I think this should be `ExpectWithOffset` , right ?

Reply:
I did not want to modify the code more than necessary as this is hot-fix for flakes. Can we follow up here?

Reply:
```suggestion
			if err != nil {
			            return nil, err
			}
```

I mean ... we can, but ... Isn't the proposed change trivial ?

Reply:
I went with `ExpectWithOffset` as I believe these are system errors and we should not defend ourselves against them.

Reply:
Fair enough.